### PR TITLE
SQL: Refactor Tableau connector to make use of the connection properties

### PR DESCRIPTION
**Pull request by bpintea**
This refactors the way the connection URL is being built, to make use of
frameworks' ability to construct the URL based on a set of connection
properties.
This comes with the benefit that the URI attributes escaping is taken
care for us.

The PR also removes the URI "manual" logging, which is already done
by the framework, with proper masking of the password.

### CODE REVIEW

**Comment by bpintea** [comment]
@elasticmachine update branch

**Comment by bpintea** [comment]
@elasticmachine run elasticsearch-ci/bwc

File: x-pack/plugin/sql/connectors/tableau/connector/connectionProperties.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    props["user"] = attr["username"];
+    props["password"] = attr["password"];
+
+    var extraProps = attr[connectionHelper.attributeWarehouse];
+    if (extraProps != null && extraProps.trim().length > 0) {
+        var avps = extraProps.trim().split(/[\s&]/);
+        for (var i = 0; i < avps.length; i++) {
+            var tokens = avps[i].split("=");
+            if (tokens.length != 2 || tokens[0].length == 0 || tokens[1].length == 0) {
+                logging.log("Malformed AVP: [" + avps[i] + "] in additional params: [" + extraProps + "]");
```
Comment (astefan) [inline x-pack/plugin/sql/connectors/tableau/connector/connectionProperties.js:13]: I guess what AVP means, but I am not sure. Especially this being a logged message, I'd give it a meaningful name. Something like "Malformed attribute-value pair:......"

File: x-pack/plugin/sql/connectors/tableau/connector/connectionBuilder.js [hunk 2]
```diff
@@ -6,35 +6,7 @@
     } else{
         urlBuilder += "http://";
     }
-    urlBuilder += attr["server"] + ":" + attr["port"];
+    urlBuilder += attr["server"] + ":" + attr["port"] + "?";
```
Comment (astefan) [inline x-pack/plugin/sql/connectors/tableau/connector/connectionBuilder.js:9]: Is the question mark at the end always necessary?

**Review by astefan — APPROVED** [review]
LGTM. I've left two comments, though

File: x-pack/plugin/sql/connectors/tableau/connector/connectionProperties.js (hunk 1 above)
Comment (bpintea) [reply x-pack/plugin/sql/connectors/tableau/connector/connectionProperties.js]: Acronym expanded.

File: x-pack/plugin/sql/connectors/tableau/connector/connectionBuilder.js (hunk 2 above)
Comment (bpintea) [reply x-pack/plugin/sql/connectors/tableau/connector/connectionBuilder.js:9]: Yes: the "user" is a mandatory config option (can't submit the form if not filled), so there will always be a property set; so adding the `?` is safe.

**Comment by bpintea** [comment]
@elasticmachine update branch

**Comment by bpintea** [comment]
@elasticmachine run elasticsearch-ci/bwc

File: x-pack/plugin/sql/connectors/tableau/connector/connectionProperties.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    var props = {};
+
+    props["user"] = attr["username"];
+    props["password"] = attr["password"];
+
+    var extraProps = attr[connectionHelper.attributeWarehouse];
+    if (extraProps != null && extraProps.trim().length > 0) {
+        var avps = extraProps.trim().split(/[\s&]/);
+        for (var i = 0; i < avps.length; i++) {
+            var tokens = avps[i].split("=");
```
Comment (matriv) [inline x-pack/plugin/sql/connectors/tableau/connector/connectionProperties.js:11]: Question, to be more lenient shouldn't we trim the split tokens, to treat something like:
` param1  =   value1  `

**Comment by bpintea** [comment]
@elasticmachine update branch

File: x-pack/plugin/sql/connectors/tableau/connector/connectionProperties.js (hunk 3 above)
Comment (bpintea) [reply x-pack/plugin/sql/connectors/tableau/connector/connectionProperties.js:11]: Space is accepted as a delimiter between the AVPs, as a friendlier alternative to `&`; i.e. both forms `a=b&c=d` and `a=b c=d` are permitted. Spaces would normally need to be encoded in a URI.
That's why the initial AVP splitting is done by `.split(/[\s&]/)`.
I've added a comment.

Comment (matriv) [reply x-pack/plugin/sql/connectors/tableau/connector/connectionProperties.js:12]: Thx for explaining!

**Review by matriv — APPROVED** [review]
LGTM

**Pull request merged**